contracts/middle-layer/GreenfieldExecutor.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "../Config.sol";
+import "../interface/ICrossChain.sol";
+import "../interface/IGreenfieldExecutor.sol";
+
+contract GreenfieldExecutor is Config, Initializable, IGreenfieldExecutor {
+    enum MsgType {
+        Unspecified,
+        CreatePaymentAccount,
+        Deposit,
+        DisableRefund,
+        UpdateParams,
+        Withdraw,
+        MigrateBucket,
+        CancelMigrateBucket,
+        CompleteMigrateBucket,
+        RejectMigrateBucket,
+        UpdateBucketInfo,
+        ToggleSPAsDelegatedAgent,
+        DiscontinueBucket,
+        SetBucketFlowRateLimit,
+        CopyObject,
+        DiscontinueObject,
+        UpdateObjectInfo,
+        LeaveGroup,
+        UpdateGroupExtra,
+        SetTag,
+        CancelUpdateObjectContent
+    }
+
+    constructor() {
+        _disableInitializers();
+    }
+    function initialize() public initializer {}
+
+    function execute(MsgType[] calldata _msgTypes, bytes[] calldata _msgBytes) external payable override returns (bool) {
+        uint256 _length = _msgTypes.length;
+        require(_length > 0, "empty data");
+        require(_length == _msgBytes.length, "length not match");
+
+        (uint256 relayFee, ) = ICrossChain(CROSS_CHAIN).getRelayFees();
+        require(msg.value == (relayFee * _length), "not enough value");
```

Comment:
I think we dont have to mul `_length`, because it is only one tx on gnfd.

Reply:
Done